product_extended_segmentation/model/template.py

```diff
@@ -247,10 +237,82 @@ def _factor(factor, product_efficiency, product_rounding):
             tmpl_obj.write(
                 cr, uid, [product_tmpl_id.id], {'standard_price': price},
                 context=context)
-        tmpl_obj.write(
-            cr, uid, [product_tmpl_id.id], sgmnt_dict, context=context)
+        return self.ensure_change_price_log_messages(cr, uid, vals, context)
+
+    @api.multi
+    def _evaluate_threshold(self, bottom_th, computed_th, diff):
+        """
+        Checks if bottom threshold is reached by computed threshold
+        """
+        precision_id = self.env['decimal.precision'].precision_get('Account')
+        return float_is_zero(diff, precision_id) or\
+            (self.standard_price and diff < 0 and
+             computed_th > bottom_th)
+
+    def _compute_threshold(self, std_price, new_price):
+        """
+         Computes price variation used for threshold checking
+        """
+        return std_price and abs((new_price - std_price) * 100 / std_price)\
+            or 0.0
+
+    @api.multi
+    def get_bottom_price_threshold(self):
+        """
+        Returns bottom threshold giving priority to product's company, if not,
+        user's company is taken
+        """
+        user = self.env['res.users'].browse(self._uid)
+        threshold = self.company_id.\
+            std_price_neg_threshold
+        if not threshold:
+            threshold = user.company_id.std_price_neg_threshold
+        return threshold
 
-        return price
+    @api.model
+    def ensure_change_price_log_messages(self, vals):
+        """
+        Write in products (unless log_only is True) and then log messages
+        if segments or products (or both) were updated
+        """
+        sgmnts_dict = vals.get('sgmnts_dict', False)
+        log_only = vals.get('log_only', False)
+        new_price = vals.get('new_price', False)
+        prod_ids = vals.get('product_ids', False)
+        if prod_ids._name == 'product.template':
+            prod_ids = prod_ids.product_variant_ids
+
+        if not prod_ids:
+            return False
+
+        # Just writting segments to be consistent with segmentation
+        # feature. TODO: A report should show differences.
+        if sgmnts_dict:
+            if not log_only:
+                prod_ids.write(sgmnts_dict)
+            subject = "Segments updated. "
+            body = 'Segments were written CHECK AFTERWARDS.'\
+                '{0}'.format(str(sgmnts_dict))
+        else:
+            body = ''
+            subject = "Segments NOT updated. "
```

Comment:
put body and subject outside and before the if, even if the else looks like is right the proper way is initialize the variables with its default value and then execute the change over them (in favor of readability):

```python
body = 'Default Body'
subject = 'Subject default'
if something:
   ....
   body = new_body
   subject = new_subject
```